Client/core/agent.py

```
@@ -0,0 +1,158 @@
+                                  self.consecutive_errors, self.max_consecutive_errors, e)
+
+                if self.consecutive_errors >= self.max_consecutive_errors:
+                    logging.error("Max consecutive errors reached. Triggering shutdown.")
+                    self.stop()
+                    break
+
+                backoff_delay = self.base_backoff * (2 ** min(self.consecutive_errors, 6))
+                logging.info("Waiting %.2f seconds before retry...", backoff_delay)
+                time.sleep(backoff_delay)
+
+    def _parse_response(self, text):
```

Ruff (0.15.0)

[warning] Missing return type annotation for private function `_parse_response`

(ANN202)